### PR TITLE
NAS-132813 / 25.04 / Convert `pool.snapshottask` to new API

### PR DESCRIPTION
**Pull request by creatorcary**
When using `AfterValidator`, the passed callable's return value is assigned to the field. For this reason, our validators should return the value that is passed to them or some modified version of the value. This change should not cause any issues since we have not been using the validators' return values until now.

Tests: http://jenkins.eng.ixsystems.net:8080/job/tests/job/api_tests/2141/

### CODE REVIEW

File: src/middlewared/middlewared/api/v25_04_0/pool_snapshottask.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+class PoolSnapshotTaskDeleteResult(BaseModel):
+    result: Literal[True]
+
+
+class PoolSnapshotTaskMaxCountArgs(BaseModel):
+    pass
+
+
+class PoolSnapshotTaskMaxCountResult(BaseModel):
+    result: Literal[512]
```
Comment (themylogin) [inline src/middlewared/middlewared/api/v25_04_0/pool_snapshottask.py:91]: Why can't it be just `int`?

File: src/middlewared/middlewared/api/v25_04_0/pool_snapshottask.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+
+class PoolSnapshotTaskMaxTotalCountArgs(BaseModel):
+    pass
+
+
+class PoolSnapshotTaskMaxTotalCountResult(BaseModel):
+    result: Literal[10000]
```
Comment (themylogin) [inline src/middlewared/middlewared/api/v25_04_0/pool_snapshottask.py:99]: Same here

**Review by themylogin — COMMENTED** [review]
Also I don't understand why do you want to use that validator module (it'll be eventually removed) and change it so massively. Let's just re-implement that field type like we did with `LocalUsername`, `LocalGID`. Also, let's put it into base types since other classes will also use it.

File: src/middlewared/middlewared/api/v25_04_0/pool_snapshottask.py (hunk 1 above)
Comment (creatorcary) [reply src/middlewared/middlewared/api/v25_04_0/pool_snapshottask.py:91]: If it makes it easier to change the number in the future then maybe it should be. I just like high specificity for the documentation.

**Comment by creatorcary** [comment]
I was under the impression that we wanted to bring our old validators into the new API so I'll correct this.

**Review by themylogin — APPROVED** [review]
Thank you!

**Pull request merged**